### PR TITLE
Bug/795 clickable product descriptions

### PR DESCRIPTION
**Pull request by AllenBW**

### CODE REVIEW

File: src/client/app/components/product-description/product-description.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <div class="product-description">
   <div ng-if="::!vm.product.img" class="product-description__image product-description__image--missing"></div>
   <img ng-if="::vm.product.img" class="product-description__image" ng-src="images/assets/{{ ::vm.product.img }}"/>
-  <span class="product-description__link" ng-click="vm.clicked()">{{ ::vm.product.name }}</span>
+  <span class="product-description__link"><a ui-sref="{{vm.linkTo}}">{{ ::vm.product.name }}</a></span>
```
Comment (stackus) [inline src/client/app/components/product-description/product-description.html]: Can we skip the span tag entirely?

File: src/client/app/resources/product.factory.js [hunk 2]
```diff
@@ -7,7 +7,18 @@
   /** @ngInject */
   function ProductFactory($resource) {
     var Product = $resource('/api/v1/products/:id', {id: '@id'}, {
-      update: {method: 'PUT'}
+      // Get Single
+      get: {
```
Comment (stackus) [inline src/client/app/resources/product.factory.js]: `get` and `query` are default routes. There is no need to ever* define them.

*: Unless you're using an api that is counter to REST principles

File: src/client/app/resources/service.factory.js [hunk 3]
```diff
@@ -6,7 +6,20 @@
 
   /** @ngInject */
   function ServicesFactory($resource) {
-    var Service = $resource('/api/v1/services/:id', {id: '@id'}, {});
+    var Service = $resource('/api/v1/services/:id', {id: '@id'}, {
+      // Get Single
+      get: {
```
Comment (stackus) [inline src/client/app/resources/service.factory.js]: `get and `query` ditto here.

File: src/client/app/states/marketplace/marketplace.html [hunk 4]
```diff
@@ -23,5 +23,8 @@
     </div>
   </div>
 </content-header>
-<catalog-category ng-repeat="category in vm.catalog" category="category" view-mode="vm.viewMode"
+<catalog-category ng-repeat="category in vm.catalog"
+                  category="category"
+                  view-mode="vm.viewMode"
+                  link-to="products.details"
```
Comment (stackus) [inline src/client/app/states/marketplace/marketplace.html]: Aren't you nixing this parameter?

File: src/client/app/states/products/details/details.html [hunk 5]
```diff
@@ -0,0 +1,64 @@
+<content-header short="true">
+  <div class="product-details-heading">
+    <img ng-if="vm.product.img" class="product-details-heading__icon" ng-src="images/assets/{{ ::vm.product.img }}"
+         fallback-image="images/no_image.png"/>
```
Comment (stackus) [inline src/client/app/states/products/details/details.html]: What directive is giving us `fallback-image`?

Comment (AllenBW) [inline src/client/app/states/products/details/details.html]: no idea, it was common with service details, can nix both if its a dead
carryover

File: src/client/app/states/products/products.state.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    routerHelper.configureStates(getStates());
+    navigationHelper.navItems(navItems());
+    navigationHelper.sidebarItems(sidebarItems());
+  }
+
+  function getStates() {
+    return {
+      'products': {
+        url: '/product',
+        redirectTo: 'products.details',
```
Comment (stackus) [inline src/client/app/states/products/products.state.js]: Might be better to redirect them to marketplace instead.

**Pull request merged**